src/MeshKernelApi/MeshKernel.cpp

```diff
@@ -76,7 +76,7 @@ namespace meshkernelapi
     static int meshKernelStateCounter = 0;
 
     // Error state
-    static char exceptionMessage[512] = "";
+    static std::string exceptionMessage{};
```

Comment:
We have discussed this in the past. I believe std::string  has a char array that can be reallocated. having a static string does not guarantee that the underlying char array is also static (no reallocation of the underlying char array occurs). We want the exceptionMessage to be allocated in the same memory address during the execution of the library. This memory address is accessed by the client, and we do not want the client to access a memory containing something else.

Reply:
I was trying to avoid the usage of strcpy, because it's very unsafe. Can you think of a better solution? memcpy comes to mind.

Reply:
Used memcpy. This way the length of the source message cannot exceed the length of the destination. Up to 512 chars can now be copied from source to destination. If the length of the source is longer, the message is truncated.